pkg/agent/controller/networkpolicy/reconciler.go

```diff
@@ -290,26 +315,67 @@ func (r *reconciler) Reconcile(rule *CompletedRule) error {
 	return ofRuleInstallErr
 }
 
+func (r *reconciler) getRuleType(rule *CompletedRule) ruleType {
+	if !r.multicastEnabled {
+		return unicast
+	}
+	for _, service := range rule.Services {
+		if service.IGMPType != nil {
+			return igmp
+		}
+	}
+
+	for _, ipBlock := range rule.To.IPBlocks {
+		ipAddr := ip.IPNetToNetIPNet(&ipBlock.CIDR)
+		if ipAddr.IP.IsMulticast() {
+			return multicast
```

Comment:
ditto

Reply:
added in documentation, it is in another pr with e2e which is still in process.